src/Sylius/Bundle/CoreBundle/Validator/Constraints/HasEnabledEntityValidator.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\CoreBundle\Validator\Constraints;
+
+use Doctrine\Common\Persistence\ManagerRegistry;
+use Doctrine\Common\Persistence\ObjectManager;
+use Symfony\Component\PropertyAccess\PropertyAccess;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\ConstraintDefinitionException;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+
+/*
+ * @author Gustavo Perdomo <gperdomor@gmail.com>
+ */
+class HasEnabledEntityValidator extends ConstraintValidator
+{
+    /**
+     * @var ManagerRegistry
+     */
+    protected $registry;
+    /**
+     * @var \Symfony\Component\PropertyAccess\PropertyAccessor
+     */
+    protected $accessor;
+
+    /**
+     * @param ManagerRegistry $registry
+     */
+    public function __construct(ManagerRegistry $registry)
+    {
+        $this->registry = $registry;
+        $this->accessor = PropertyAccess::createPropertyAccessor();
```

Comment:
Then you can inject a different implementation of `PropertyAccessorInterface` and this instance will be reused through entire application (no need to create new property accessor if you already have one `property_accessor`).